app/models/user.rb

```diff
@@ -0,0 +1,28 @@
+class User
+  include Mongoid::Document
+  include ActiveModel::SecurePassword
+  include ActiveModel::Validations
+
+  field :email, type: String
+  index({email: 1}, {unique: true, name: 'email_unique_index'})
+  field :login, type: String
+  index({login: 1}, {unique: true, name: 'login_unique_index'})
+  field :password_digest, type: String
+  field :name, type: String
+
+  has_secure_password
+
+  validates :password, length: {in: 6..20}, presence: true
+  validates :email, uniqueness: true, format: {
+      with: /\A(|(([A-Za-z0-9]+_+)|([A-Za-z0-9]+\-+)|([A-Za-z0-9]+\.+)|([A-Za-z0-9]+\++))*[A-Za-z0-9]+@((\w+\-+)|(\w+\.))*\w{1,63}\.[a-zA-Z]{2,6})\z/i,
+      message: 'invalid email format'
+  }, allow_blank: true
+  validates :login, uniqueness: true, length: {minimum: 3}, allow_blank: true
+  validates :name, length: {minimum: 3}, allow_blank: true
```

Comment:
Overall, I don't recommend really building strict validation rules. `Yi` is pretty valid name in my opinion, but this way it won't fit. 

It actually applies to all personal information - it is better to not validate at all and use raw data, then apply counterfeit validations when there are no clearly defined rules from analytics or product owners.